test/e2e/storage/csi_volumes.go

```diff
@@ -331,6 +364,21 @@ func startPausePod(cs clientset.Interface, t testsuites.StorageClassTest, ns str
 	return class, claim, pod
 }
 
+// getDriverDaemonSet finds *any* DaemonSet in framework's namespace and returns it.
+// It must be called just after driver installation, where the only DaemonSet in the
+// namespace must be the driver.
+func getDriverDaemonSet(f *framework.Framework) (*appsv1.DaemonSet, error) {
```

Comment:
this function makes some significant assumptions:
1) Driver is deployed with Daemonset
2) Driver is deployed with *strictly* 1 Daemonset
3) No other non-driver Daemonsets deployed

I am not sure these are reasonable assumptions to make for arbitrary drivers

Reply:
Should it work for arbitrary drivers? I expect it to be used only in kubernetes e2e to test the Kubernetes part. I expect that each driver will have its own assumption how and when it can be updated.

If some generic mechanism is required (is it?), I can implement new `TestDriver` method to update a driver.

Reply:
maybe it doesn't have to but its worth calling out in a comment, any function written might get "resused" in some other code without true understanding of what it does just based on the function name.

Reply:
This part is failing on our external GCE PD tests as we bring up one driver in the default namespace as opposed to one in every test namespace. I wonder if maybe we can get the daemonset from a config specified namespace instead of using the test namespace. What do you think?

Also this test should be marked `[Serial] [Disruptive` if we make that change since it may affect usage of a shared driver.

/cc @msau42

Reply:
Yes those sound like good enhancements

Reply:
https://github.com/kubernetes/kubernetes/pull/71527

Reply:
Did not mark as serial or disruptive because the k8s tests are not disruptive. And I don't think the external tests respect the "Disruptive" flag anyway (?). We should make sure the external CI is running tests serially anyway because of this test